runtime/src/builtin_programs.rs

```diff
@@ -16,13 +16,30 @@ impl BuiltinProgram {
     }
 }
 
-pub fn get_builtin_programs() -> Vec<BuiltinProgram> {
+fn new_system_program_activation_epoch(operating_mode: OperatingMode) -> Epoch {
+    match operating_mode {
+        OperatingMode::Development => 0,
+        OperatingMode::Preview => std::u64::MAX / 2,
+        OperatingMode::Stable => std::u64::MAX / 2,
+    }
+}
+
+/// All builtin programs that should be active at the given (operating_mode, epoch)
+pub fn get_builtin_programs(operating_mode: OperatingMode, epoch: Epoch) -> Vec<BuiltinProgram> {
     vec![
-        BuiltinProgram::new(
-            "system_program",
-            system_program::id(),
-            system_instruction_processor::process_instruction,
-        ),
+        if epoch < new_system_program_activation_epoch(operating_mode) {
+            BuiltinProgram::new(
+                "system_program",
+                system_program::id(),
+                system_instruction_processor::process_instruction, // TODO: This is the OLD system instruction processor
```

Comment:
Big advantage to this approach IMO is that `system_instruction_processor` doesn't need to be peppered with `if` checks.   We make a copy of `system_instruction_processor.rs` into `legacy_system_instruction_processor.rs` with the old code, then `system_instruction_processor.rs` only contains the new code.   Now devs don't need to dance around `if`s as they make future modifications to `system_instruction_processor.rs`

Reply:
Instead of "legacy" how about starting off with a version in case there are more of these kinds of changes later?

Reply:
Once the new `system_instruction_processor` is deployed to all clusters I was thinking we delete `legacy_system_instruction_processor` from master and then we're back to just `system_instruction_processor`.  But yes in that time it means we could ends up with a `legacy_legacy_instruction_processor`.    

I like the idea of keeping the latest version just named `system_instruction_processor`. maybe `legacy_system_instruction_processor0` instead of `legacy_system_instruction_processor` so there's room to add `legacy_system_instruction_processor1` if needed?

Reply:
I went with `legacy_system_instruction_processor0`